### ISSUE TITLE
Permit default other than None for ModelChoicePreference

### ISSUE THREAD

**Issue by leetucker** [EliotBerriot/django-dynamic-preferences#88]
Right now, if one is using a ModelChoicePreference, it is impossible to set anything other than `default = None`.  If the default is set to anything else, then an initial migration will fail, because the tables do not yet exist for the default to be populated or queried.

For example, my project contains a categorical model (ExpenseCategory) that is used to describe instances of another model (ExpenseItem) via a foreign key relationship.  Since there are certain types of ExpenseItems that are automatically generated within the project, I would like to be able to automatically create ExpenseCategory items, store a reference to each category as a dynamic preference, and then reference the preference when generating expense items in order to place them in the appropriate category.

An example would be as follows.  In `dynamic_preferences_registry.py`:

	from dynamic_preferences.registries import global_preferences_registry
	from dynamic_preferences.types import ModelChoicePreference
	from .models import ExpenseCategory

    ...

	@global_preferences_registry.register
	class DefaultExpenseCat(ModelChoicePreference):
    	section = financial
    	name = 'defaultExpenseCategory'
	    verbose_name = 'Default Expense Category'
	    model = ExpenseCategory
	    default = ExpenseCategory.objects.get_or_create(name='Default Category')

Elsewhere:
    
	from dynamic_preferences.registries import global_preferences_registry
	from .models import ExpenseItem

	# We instantiate a manager for our global preferences
	global_preferences = global_preferences_registry.manager()

	# Create an expense item in the default category
	ExpenseItem.objects.create(
        amount=10,
        category=global_preferences['financial__defaultExpenseCategory'],
    )

Right now, the above code fails on initial migration with `django.db.utils.OperationalError: no such table: financial_expensecategory`

I imagine that there are probably several creative ways to get around this limitation at present, but since this seems like a common use case for the dynamic_preferences app, I figured that it should be reported as a feature request.

**Comment by leetucker** [comment]
OK, so it appears that there is a pretty straightforward way to implement this using the existing codebase.  I did so by defining a `get_default()` method for each instance, which handles the `get_or_create()` call the first time that the preference value is requested.  My app's dynamic_preferences_registry.py now reads as follows:

    from django.db import connection
    from dynamic_preferences.registries import global_preferences_registry
    from dynamic_preferences.types import ModelChoicePreference
    from .models import ExpenseCategory
    
    ...
    
    @global_preferences_registry.register
    class DefaultExpenseCat(ModelChoicePreference):
	section = financial
	name = 'defaultExpenseCategory'
        verbose_name = 'Default Expense Category'
        model = ExpenseCategory
    
        def get_default(self):
            if self.model and self.model._meta.db_table in connection.introspection.table_names:
                return ExpenseCategory.objects.get_or_create(name='Default Category')[0]

Perhaps there is a slightly nicer way to implement this, but sinc this is actually fairly straightforward, it might also be worthwhile to leave it as is and simply document the use of this model method for this use case.

**Comment by EliotBerriot** [comment]
Yep this part of the API feels a bit clunky. What happens if you get rid of the introspection part?

**Comment by leetucker** [comment]
OK, it looks from my testing like the introspection is not actually necessary.  I think that I was confused on that, because I was trying multiple approaches to get around the inability to set the default parameter directly, and the ability to define `get_default()` is I believe undocumented.

In any case, I think that it is not possible to define a default using get_or_create() like this without invoking the `get_default()` method, because the `get_or_create()` call gets executed on class initialization, which is before other apps' models may be initialized.  So, it seems to be that the two options would be:
1. Document this usage of `get_default()` to set defaults for ModelChoiceField instances, or
2. Define some other method that (for example) takes the parameters that would go into a `get_or_create()` call and uses them to define a default the first time that the preference is loaded into the database, but does not execute anything subsequently.

You would know this better than me, but I think that the method that I'm using may end up involving extra DB calls every time that a preference form is loaded, so that might argue in favor of option 2.  On the other hand, this is a pretty simple workaround, so maybe it's better to leave it as is and just document it.
